Add up all external incentive APYs for a specific duration

srph opened the pull request:
In continuation of #886, this adds up all external incentive APYs for the given duration rather than just calculating it from the first incentive.

![image](https://user-images.githubusercontent.com/5093058/197250612-63a3474b-23dd-4f17-8263-3720cb7c4777.png)

jonator commented:
Actually- can we change it to find the incentives for a specific external gauge (given an ID), rather than adding them together for the duration?

Our new design breaks down the APRs by gauge:
<img width="318" alt="Screen Shot 2022-10-21 at 3 38 43 PM" src="https://user-images.githubusercontent.com/4606373/197284747-dcc84cf7-6c22-49ed-a37d-29d7b6bb349f.png">

srph commented:
@jonator Some updates:

- Replaced allowedGauges with gaugeId and denom. I wasn't sure how to get the denom without having to explicitly pass it as a parameter.
- Moved sum computation in the pool page component.

Let me know if there's anything I may have missed. Thanks!

jonator commented on packages/stores/src/queries/pool-incentives/incentivized-pools.ts:
```diff
@@ -121,49 +123,52 @@ export class ObservableQueryIncentivizedPools extends ObservableChainQuery<Incen
    * Computes external incentive APY for the given duration
    */
-  readonly computeExternalIncentiveAPYForSpecificDuration = computedFn(
+  readonly computeExternalIncentiveGaugeAPR = computedFn(
```
Nitpick: can we please update the comment for this function as well to use APR?

jonator reviewed (changes requested):
Looks great!

jonator commented on packages/stores/src/queries/pool-incentives/incentivized-pools.ts:
```diff
@@ -119,51 +121,54 @@ export class ObservableQueryIncentivizedPools extends ObservableChainQuery<Incen
+      const observableGauge = this.queryGauge.get(gaugeId);
+
+      if (
+        duration.asMilliseconds() !==
```
I'm realizing that since we take in the gaugeId, there's no real reason to take in the duration as well and perform this check. Can we remove the duration from the params?

srph replied:
@jonator Good catch, working on it =D

jonator replied:
Please make sure the comment/docs is updated as well.

srph replied:
I think the main reason why we need duration is - on the pool page - we're iterating all through the `ExternalIncentiveGaugeAllowList[pool.id]`. Would you like me move duration checking there?

jonator reviewed (changes requested):
Sorry noticed one more thing. Should be good after that.

srph commented:
Let me know if there's anything else I might have missed @jonator

jonator reviewed (approved):
Great work! Appreciate it
